### DISCUSSION TITLE
Limited proprietary license with access to font sources

### DISCUSSION THREAD

**Discussion by amerlyq** [fabrizioschiavi/pragmatapro#196] [Ideas]
## Problem:
There are many opened issues about this font on github, and some more "vague" unpublished ones. 
They irritate me personally till the point I want to invest my time and fix them personally -- because I'm not satisfied with current fixup speed and absent milestones.
I understand, that by "buying" font we bought "font + some upgrades" **only** -- and no support was promised (despite overall community unspoken naivety).

## Prior art:
Open-source way to solving maintenance problems is _outsourcing_ them to community (e.g. github) with specific licensing terms, so everybody could contribute "where they feel personal pain the most".
I know, @fabrizioschiavi, you tried to do that decade ago, unsuccessfully (which is a pity and an understandable result due to plague of IT mentality).

## Future:
I don't really know how much more you will have enthusiasm to maintain your own font on your own.

But for me personally it would be a great loss, if font become stagnant and die -- because I will be forced to adapt to FOSS alternatives like FiroCode -- and I don't like changes, don't like compromises, and can't live with small everyday irritations.

So I would like to resurrect the discussion of going FOSS -- at least at some point in future, so it could live on and do not become outshined and buried by existing FOSS alternatives. And I could continue use it for next 20+ years, fixing small irritations here and there. It would be the best variant, but if not -- see below.

## Proposal:
Considering going FOSS is still out of question, will you be interested in providing **partial proprietary license on font source code to your customes**, @fabrizioschiavi ?

So at least whoever bought your font could make changes, make PRs against your private github repo, and you merged and shared it again with other customes -- like proper maintainer would do, literally lifting up part of your burden from github issue tracker?
I don't mind signing up Contribution Agreement reassigning the rights back to you on all my changes, I only want fix being done, nothing more.

## Considerations:
Of course, sharing sources even privately may (and will) result in source piracy, and will spawn dozens of copycats, and arise stealing IP into other FOSS and proprietary fonts, and nobody *will buy your font ever again (c)*.
They will be compiling it from available source instead of buying (like synergy(1) ) -- downloading from your private site, your public github account or by some pirate means -- whatever you decide as means of distribution.

...And still it won't be that much different from what we have now. Number of future customers won't deviate that much.
After all, I bought the font not because there are no pirate versions at all, or that I couldn't get updated version.
There are many copies available, mind me -- and I used them as a preview for at least a year or two before deciding I can't replace your font by anything else :)
So, I finally bought font because I had enough money and my consciense was dirtied by licensing terms -- literally I granted you money because I wished to give you money and restore my "lawfulness", not because there were no other ways. 
It's similar to what @MauganRa earlier proposed you about Patreon -- people simply giving money to whomever they like, simply because they able to and feel like it.

And it's almost the same with source code.
Whoever is able and wishes to buy -- will buy font nevertheless: as appreciation of your efforts or as means to clear your license terms.
So, how does it sound -- to provide at least **limited proprietary license to your source code**, if going FOSS is out of question?

**Comment by fabrizioschiavi** [comment]
Thank you @amerlyq for sharing your thought!
I’m very busy because I'm working at very time spending new projects and I didn't found the time to work at the new releases of PragmataPro.
PragmataPro is not a dead project at all. A lot of improvements are planned.
I can't change the license into FOSS because it helps me and my family to live.
I know, a lot of time passed from the latest update but I ask you to have a bit more of patience. Thank you!

**Comment by amerlyq** [comment]
If going FOSS (notice absense of "L")  is out of questions, let's enumerate only essential points of above proposal:
* I have "pains" with PragmataPro font, which are not addressed by product owner for months (yeah, I know, original price didn't include customer support and SLA)
* I'm ready to spend **my own time** to address my own pains, and **share fixes with you** and your customers, which you may accept or ignore in mainstream font distribution
* I'm ready to sign EULA and CLA and reassign my rights of these patches back to you (like Qt and other shady opensource projects already do)
* But to be able to contribute I need private access to sources and build instructions, of course it's the main problem for you, considering previous history of PragmataPro binary distributions.

#### So what do you think about above short proposal?
-----

Considerations
* In general providing such "private access" results in the culture of *intrasource* when small closed community, literally "for free and on their own accord" improves soft of someone getting all the money (new popular cult/trend in big companies, if you remove all the fancy words around it), **but I'm OK with it** -- of course until I'm able to build font with my changes for myself
* I consider it's not only me, but some people from this issue list will be happy to do the same -- so it raises a question of providing us a way to do it legally
* In programmers' world library vendors often have different legal contracts for distributions provided in binary and in source forms, sometimes with different prices, so it's not something out of questions, if you provide sources for additional reasonable sum per license (however, in this case, depending on the price, desire to contribute back may significantly cool down)
* Considering you provide these sources with "standard commercial license" like e.g. [WolfSSL does](https://www.wolfssl.com/license/) there is no way your sources will be incorporated into any MIT or GPL products from FOSS world, even if sources will be leaked by someone non-trustworthy (these communities aren't so shameless and definitely not above the law)
* The only concern is the "corporate piracy" -- for which you (most likely) won't be able to sue them (due to your limited resources). But I'm sure "font sources" in this case is the least of concerns -- if they reimplement/steal your font typefaces even 1:0.98 *visually/feel*, you will be dissatisfied even if they used different sources.

And back to repeated rantings about possible FOSS in future:
* Of course, the real problem here is "the loss of potential income the font sometimes generates". I'm really doubtful there is much correlation between number of people ready to buy your font and having leaked, often obsolete, sources. I talked about this situation in the first post of this tread.
* Those who wish to buy legally -- will buy it for different reasons than "unavailabilty". Most likely for moral/law reasons.
* Those who wish to have access to the latest fixes -- will buy anyway to address their constant "pains" (and font is still long road from being ideal). Otherwise they would pirate some binary already present in web and won't buy it in the first place under any condition. 
* Those who aren't tech-savvy enough to build font themselves -- will buy to have binary distribution built by someone for them, and better it be official.
* Those who **are** tech-savvy to work with font sources -- most likely will go to other FOSS fonts for ideological reasons, or pirate it from sources and won't buy it anyway, because they don't have money. 
* I.e. you got the same overall amount of money, the society is ready to pay you for your work in both cases, but got much wider adoption in case of "openness" -- which opens ways to incorporate community changes back into font through CLA.
* Those of IT world who have money but decide to pirate font anyway -- are most likely treating any IT work as trash and despite earning money from IT themselves -- don't value and don't wish to pay for similar work of others. This is contradictory, but's it's the present mental distortion of most people of current generations. They **won't pay anyway**.

Basically, what you provide to people wishing to buy PragmataPro *now* (in 2020) is not the "product" (in its early days sense) but paid service of its future improvement (by yourself), *sense of belonging* to its users community -- for its buyers, and *sense of gratitude/justice/merit* these buyers experience when actually, well... buying. IMO I hardly see anything else, having value over pirate copy, but anyone may complement my list. So, there are two roles here: "paid further development" and "original author royalties".
The questions are: 
1) how much does *this* (service) costs from your perspective;
2) and is the current model of ensuring revenue still works now and in future?

War of having totally controlled distribution of "product": to be paid for each copy, was partially lost with first ever pirate copy of the font appearing on the WWW (it was inevitable) -- and gains of this business model become more and more marginal over time. Nowadays you must be the billion-revenue "Record Label" or "Publisher" to hire legal team, lobby your laws, hire dev team, write bot scrapers to go over all sites to fill-in DMCA and sue everybody for any resistance. It doesn't work for individuals.

The only "additional value" which still generates revenue over "unique value" (best typefaces for some category of users) -- is the continuous improvements you provide to font. But you are the only one developer/designer/contributor -- with limited time and energy. Which is inevitably dries out over time -- together with community interest, due to owner "responsiveness" -- which feels less and less. So the revenue will inevitably go down withouth constant "live code" and new blood flowing in to sustain exposure -- both for potential buyers and people looking forward to make changes.

-----

Alternatively, @all is there any nice guide to patch fonts in binary form, preferably with some typical cases walk-throughs? It won't help to cover all my pains, but at least alleviate some of them temporarily.
